Functions/xASL_quant_HadamardDecoding.m

```diff
@@ -26,12 +28,22 @@
 % __________________________________
 % Copyright 2015-2021 ExploreASL
 
-% Decoding Fields
+% Checking if all inputs are present
 
-if nargin<1 || isempy(Encoded_ASL)
-    
+if nargin<3 && isempty(Encoded_ASL)
+    warning('Encoded_ASL input is empty');
+end
+
+if nargin<3 && isempty(xDecodingFields)
+    warning('xDecodingFields input is empty');
 end
 
+if nargin<3 && isempty(NumberEchoTimes)
+    warning('NumberEchoTimes input is empty');
+end
```

Comment:
@BeatrizPadrela I think I'm distracting you too much 😆 try nargin<1, nargin<2, ... 😉